bluepysnap/schemas/schemas.py

```diff
@@ -206,3 +208,65 @@ def validate_edges_schema(path, edges_type, virtual):
     errors = _validate_schema_for_dict(_parse_schema("edge", edges_type), edges_h5_dict)
 
     return _wrap_errors(path, errors, "/")
+
+
+def _resolve_types(resolver, types):
+    """Use jsonschema `resolver` to resolve the `types` dict."""
+    cache = {}
+
+    def _resolve_type(type_):
+        if type_ not in cache:
+            type_ = resolver.resolve(type_)[1]["properties"]["datatype"]["const"]
+            if hasattr(np, type_):
+                cache[type_] = getattr(np, type_)
+            elif type_ == "utf-8":
+                cache[type_] = str
+
+        return cache[type_]
+
+    return {k: _resolve_type(v["$ref"]) for k, v in types.items()}
+
+
+def nodes_schema_types(nodes_type):
+    """Get the datatypes of the attribute for nodes.
+
+    Args:
+        nodes_type (str): node type (e.g., "biophysical")
+
+    Returns:
+        dict: name -> type of column
+    """
+    schema = _parse_schema("node", nodes_type)
+    resolver = jsonschema.validators.RefResolver("", schema)
+
+    schema = schema["$node_file_defs"]["nodes_file_root"]["properties"]["nodes"]
+    schema = schema["patternProperties"][""]["properties"]["0"]["properties"]
+    dynamics_params = schema["dynamics_params"]["properties"]
+    del schema["dynamics_params"]
+    del schema["@library"]
+
+    return _resolve_types(resolver, schema), _resolve_types(resolver, dynamics_params)
+
+
+def edges_schema_types(edges_type, virtual):
+    """Get the datatypes of the attribute for nodes.
+
+    Args:
+        nodes_type (str): node type (e.g., "chemical")
+        virtual(bool): whether this a virtual edge population
+
+    Returns:
+        dict: name -> type of column
+    """
+    if virtual:
+        edges_type += "_virtual"
+
+    schema = _parse_schema("edge", edges_type)
+    resolver = jsonschema.validators.RefResolver("", schema)
+
+    schema = schema["$edge_file_defs"]["edges_file_root"]["properties"]["edges"]
```

Comment:
not super happy about having to dig so deep into the dictionary - if you have a better idea on how to do this, that would be cool.

Reply:
Yeah, I get it. 

If the use case is just to get a mapping of a node/edge property to a dtype (i.e., we don't actually care about the edge/node type), we could have the properties of `<edge|node>/<population_name>/0` in their own files. Something like `{edge,node}_0_properties.yaml`.

And then just add a reference to those in the `{edge,node}.yaml` files.

That would kill two birds with one stone: 
* make it simpler to do the parsing of the properties here
* make the `{edge,node}.yaml` files more readable

I'm also not super happy of the long list of properties, or rather the fact that we need to do:
```yaml
property_1:
     $ref: dtype_1
property_2:
     $ref: dtype_1
...
```
if we could have something like
```yaml
dtype_1:
  - property_1
  - property_2
 ...
```
it would be cleaner, IMO. Not sure if this is easily doable but I'll give it some thought. For sure it is, if we make some custom parsing before feeding it to `jsonschema` but that is the route I'd rather avoid if possible.

Reply:
I guess we can do this reorg some other time; I understand better now the structure, and why it is like it is.

I'm going to play w/ this a bit more, to see if it works for my use case.

Reply:
Sure. fine with me